src/map/entities/battleentity.cpp

```diff
@@ -1744,8 +1744,10 @@ bool CBattleEntity::OnAttack(CAttackState& state, action_t& action)
 
     /////////////////////////////////////////////////////////////////////////
     //  Start of the attack loop.
+    //  Make sure our target is alive on each iteration to not overkill;
+    //  And make sure we aren't dead in case we died to a counter.
     /////////////////////////////////////////////////////////////////////////
-    while (attackRound.GetAttackSwingCount() && !(PTarget->isDead()))
+    while (attackRound.GetAttackSwingCount() && !PTarget->isDead() && this->isAlive())
```

Comment:
kinda weird for the previous author to use "not isDead" when thats exactly the same as checking "isAlive". Both return the same true/false-ness on of the entity is alive.

Reply:
up to you if you'd like to turn both checks into 
` while (attackRound.GetAttackSwingCount() && PTarget->isAlive() && this->isAlive())`

Reply:
True, no real reason to use both. I'll change it.